tests/components/light.test.js

```diff
@@ -109,5 +118,143 @@ suite('light', function () {
+      lightEl.setAttribute('light', 'target: #othertarget');
+
+      assert.strictEqual(lightEl.getObject3D('light').target, othertargetEl.object3D);
+    });
+
+    test('spotlight: when created, light target is child object positioned at 0 0 -1', function () {
+      var lightEl = this.el;
+
+      lightEl.setAttribute('light', 'type: spot');
+
+      assert.equal(lightEl.getObject3D('light').target, lightEl.getObject3D('light-target'));
+      assert.equal(lightEl.getObject3D('light').target.position.x, 0);
+      assert.equal(lightEl.getObject3D('light').target.position.y, 0);
+      assert.equal(lightEl.getObject3D('light').target.position.z, -1);
+    });
+
+    test('spotlight: when target is changed to null, light target is child object positioned at 0 0 -1', function () {
+      var sceneEl = this.el.sceneEl;
+      var lightEl = this.el;
+      var targetEl = document.createElement('a-entity');
+
+      targetEl.setAttribute('id', 'target');
+      sceneEl.appendChild(targetEl);
+
+      lightEl.setAttribute('light', 'type: spot');
+      lightEl.setAttribute('light', 'target', '#target');
+      lightEl.setAttribute('light', 'target', null);
+
+      assert.equal(lightEl.getObject3D('light').target, lightEl.getObject3D('light-target'));
+      assert.equal(lightEl.getObject3D('light').target.position.x, 0);
+      assert.equal(lightEl.getObject3D('light').target.position.y, 0);
+      assert.equal(lightEl.getObject3D('light').target.position.z, -1);
+    });
+
+    test('directional: set light target with selector when light is created', function () {
+      var sceneEl = this.el.sceneEl;
+      var lightEl = this.el;
+      var targetEl = document.createElement('a-entity');
+
+      sceneEl.appendChild(targetEl);
+      targetEl.setAttribute('id', 'target');
+
+      lightEl.setAttribute('light', 'type: directional; target: #target');
+
+      assert.strictEqual(lightEl.object3D.children[0].target, targetEl.object3D);
+    });
+
+    test('directional: change light target with selector', function () {
+      var sceneEl = this.el.sceneEl;
+      var lightEl = this.el;
+      var targetEl = document.createElement('a-entity');
+      var othertargetEl = document.createElement('a-entity');
+
+      targetEl.setAttribute('id', 'target');
+      othertargetEl.setAttribute('id', 'othertarget');
+      sceneEl.appendChild(targetEl);
+      sceneEl.appendChild(othertargetEl);
+
+      lightEl.setAttribute('light', 'type: directional');
+      lightEl.setAttribute('light', 'target: #target');
+      lightEl.setAttribute('light', 'target: #othertarget');
+
+      assert.strictEqual(lightEl.object3D.children[0].target, othertargetEl.object3D);
+    });
+
+    test('directional: when created, light target is positioned at 0 0 0, but NOT a child', function () {
+      var lightEl = this.el;
+
+      lightEl.setAttribute('light', 'type: directional');
+
+      assert.typeOf(lightEl.object3D.children[0].target, 'object');
```

Comment:
You can do `lightEl.getObject3D('light')` all the other places in this file. And I would store it in a `var` to reduce clutter.